var/spack/repos/builtin/packages/relion/package.py

```diff
@@ -34,29 +35,55 @@ class Relion(CMakePackage, CudaPackage):
     homepage = "http://http://www2.mrc-lmb.cam.ac.uk/relion"
     git      = "https://github.com/3dem/relion.git"
 
-    version('develop')
-    version('2.1', tag='2.1')
-    version('2.0.3', tag='2.0.3')
+    version('3.0_beta', git='https://bitbucket.org/scheres/relion-3.0_beta.git')
+    version('2.1', git='https://github.com/3dem/relion.git', preferred='true', tag='2.1')
+    version('2.0.3', git='https://github.com/3dem/relion.git', tag='2.0.3')
+    version('develop', git='https://github.com/3dem/relion.git')
+
 
     variant('gui', default=True, description="build the gui")
     variant('cuda', default=True, description="enable compute on gpu")
     variant('double', default=True, description="double precision (cpu) code")
     variant('double-gpu', default=False, description="double precision (gpu) code")
+    # added below cluster and desktop variants given some shared libraries in /usr/lib, like gpfs,
+    # are loaded by ld.so after compile which makes relion not work on desktops which may
+    # not be a member of gpfs cluster thus lack these given libraries
+    # https://github.com/spack/spack/issues/8384
+    variant('cluster', default=True, description="build relion for use in cluster")
+    variant('desktop', default=False, description="build relion for use in desktop")
     variant('build_type', default='RelWithDebInfo',
             description='The build type to build',
             values=('Debug', 'Release', 'RelWithDebInfo',
                     'Profiling', 'Benchmarking'))
 
     depends_on('mpi')
+    # relion will not build with newer versions of cmake
+    # per https://github.com/3dem/relion/issues/380
+    depends_on('cmake@3:3.9.4', type='build')
     depends_on('fftw+float+double')
     depends_on('fltk', when='+gui')
-    # cuda 9 not yet supported
-    #  https://github.com/3dem/relion/issues/296
-    depends_on('cuda@8.0:8.99', when='+cuda')
-    # use gcc < 5 when compiled with cuda 8
-    conflicts('%gcc@5:', when='+cuda')
+    depends_on('libtiff')
+
+    # relion 3 supports cuda 9
+    # relion < 3 does not
+    depends_on('cuda', when='+cuda')
+    depends_on('cuda@9:', when='@3: +cuda')
+    depends_on('cuda@8.0:8.99', when='@:2 +cuda')
+
+    # use gcc 4 when using cuda8 
+    # use up to gcc 6 when using cuda9
+    conflicts('%gcc@7:', when='@3: +cuda')
+    conflicts('%gcc@5:', when='@:2 +cuda')
```

Comment:
Is this a restriction on building relion or a restriction on building cuda?

Reply:
It's a cuda restriction but it was easier for me to put the conflicts in this spec

Reply:
Hi @adamjstewart 

Can these conflicts remain?

Reply:
Shouldn't they be added to the Cuda package instead of every package that depends on Cuda?